x-pack/plugins/security_solution/server/lib/machine_learning/validation.ts

```diff
@@ -20,14 +20,14 @@ export class HttpAuthzError extends Error {
   }
 }
 
-export const toHttpError = (validation: Validation): HttpAuthzError | undefined => {
+export const toAuthzError = (validation: Validation): HttpAuthzError | undefined => {
```

Comment:
should we call it `toHttpAuthzError` ?

It uses under the hood 

```
 if (!validation.valid) {
    return new HttpAuthzError(validation.message);
  }

```

which has statusCode in properties

Reply:
`toAuthzError` sounds okay as by me. It's an internal method of `machine_learning/validation.ts` and is not used anywhere outside this file.

Reply:
I also meant all exports here: they have now AuthError instead of HttpError, but under the hood there is

 ```
    return new HttpAuthzError(validation.message);
```

I think renaming this exported method to `throwAuthzError` doesn't fully reflect a fact, that underlying thrown err is `HttpAuthzError `, which contains statusCode in it. I guess renaming it as `throwHttpAuthzError` or even leaving as it was before reflects better its purpose

Reply:
I'm afraid I'm not following your logic here. We already have errors like `BadRequestError` or `NotFoundError`; they all have status codes and don't have `Http` in their name. `AuthzError` follows this suite perfectly.

Reply:
I don't think though NotFoundError is correct then.
It doesn't have 404 statusCode in it.

You can check in one of the tests and NotFound implementation

```
        message: 'Rule not found',
              status_code: 500,
```
BadRequestError has very self explaining name as well, which does not need including Http into it(it has 400 statusCode also)

But if you think it's fine to go with renaming to `AuthzError`, let's have it this way

Reply:
> I don't think though NotFoundError is correct then.
> It doesn't have 404 statusCode in it.

From what I've seen, `Rule not found` is thrown as a regular `Error`. Probably that's why it has `status_code: 500`.